lib/standards/aria-roles.js

```diff
@@ -0,0 +1,221 @@
+// Source: https://www.w3.org/TR/wai-aria-1.1/#roles
+
+/* easiest way to see allowed roles is to filter out the global ones
+   from the list of inherited states and properties. The dpub spec
+   does not have the global list so you'll need to copy over from
+   the wai-aria one:
+
+  const globalAttrs = Array.from(
+    document.querySelectorAll('#global_states li')
+  ).map(li => li.textContent.replace(/\s*\(.*\)/, ''));
+
+  const globalRoleAttrs = Array.from(
+    document.querySelectorAll('.role-inherited li')
+  ).filter(li => globalAttrs.includes(li.textContent.replace(/\s*\(.*\)/, '')))
+
+  globalRoleAttrs.forEach(li => li.style.display = 'none');
+*/
+const ariaRoles = {
+	alert: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded'],
+		nameFrom: ['author']
+	},
+	alertdialog: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded', 'aria-modal'],
+		nameFrom: ['author']
```

Comment:
Since we're changing the structure, can we get rid of this, and replace it with a `nameFromContent?: boolean`? (without the "S", ARIA calls it "name from content")

Everything is name from author, there's no reason to put that everywhere.